libs/state/src/lib/core/operators/distinctUntilSomeChanged.ts

```diff
@@ -0,0 +1,173 @@
+import { MonoTypeOperatorFunction } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+
+/**
+ * @internal
+ */
+export type CompareFn<T> = (x: T, y: T) => boolean;
+/**
+ * @internal
+ */
+export type KeyCompareMap<T extends object> = {
+  [K in keyof T]: CompareFn<T[K]>;
+};
+
+/**
+ * @description
+ *
+ * Returns an Observable that emits all items emitted by the source Observable that are distinct by comparison from
+ * the previous item. You can provide a custom comparison for each key individually by setting a `KeyCompareMap<T>`.
+ * If no comparison is provided for a specified key, an equality check is used by default.
```

Comment:
I don'T see the equality check happening in the code. 
Also, keys or keymap is a required param. 
IMHO this is wrong.

Reply:
`const defaultCompare: CompareFn<T[K]> = (oldVal, newVal) => oldVal === newVal;` this is the equality check. So this is true :)

Reply:
Ah! 
> For a key

Got it!